plugins/flashrom/fu-flashrom-lspcon-i2c-spi-device.c

```diff
@@ -246,9 +305,31 @@ fu_flashrom_lspcon_i2c_spi_device_setup (FuDevice *device, GError **error)
 	instance_id = g_strdup_printf ("FLASHROM-LSPCON-I2C-SPI\\VEN_%s&DEV_%s", vid, pid);
 	fu_device_add_instance_id (device, instance_id);
 
+	system_family = fu_context_get_hwid_value (fu_device_get_context (device),
+						   FU_HWIDS_KEY_FAMILY);
+	g_free (instance_id);
+	instance_id = g_strdup_printf ("FLASHROM-LSPCON-I2C-SPI\\HwidFamily=%s",
+				       system_family);
+	fu_device_add_instance_id_full (device, instance_id,
+					FU_DEVICE_INSTANCE_FLAG_ONLY_QUIRKS);
+
 	return fu_flashrom_lspcon_i2c_spi_device_set_version (device, error);
 }
 
+static gboolean
+fu_flashrom_lspcon_i2c_spi_device_detach (FuDevice *device, GError **error)
+{
+	return FU_DEVICE_CLASS (fu_flashrom_lspcon_i2c_spi_device_parent_class)
+		->open (device, error);
+}
```

Comment:
isn't this a bit weird? open for detach? I'm not sure how much stuff is going to explode, as sometimes we call ->open(), ->detach(), ->close() etc.

Reply:
I agree it's weird, but the alternative is to move where calls to `flashrom_programmer_init` and friends live. Given `fu_flashrom_device_open` doesn't chain up and the `open` method on the LSPCON device is changed here to avoid chaining up to `FlashromDevice`, this seems robust.